website/source/docs/internals/plugins/devices.html.md

```diff
@@ -3,10 +3,80 @@ layout: "docs"
 page_title: "Device Plugins"
 sidebar_current: "docs-internals-plugins-devices"
 description: |-
-  Learn about how to author a Nomad device plugin.
+  Learn how to author a Nomad device plugin.
 ---
 
 # Devices
 
-Device plugin documentation is currently a work in progress. Until there is
-documentation, the [Nvidia GPU plugin](https://github.com/hashicorp/nomad/tree/master/devices/gpu/nvidia) is a useful example.
+Nomad has built-in support for scheduling compute resources such as CPU, memory,
+and networking. Nomad device plugins are used to support scheduling tasks with
+other devices, such as GPUs. They are responsible for fingerprinting these
+devices and working with the Nomad client to make them available to assigned
+tasks.
+
+For a real world example of a Nomad device plugin implementation, see the [Nvidia
+GPU plugin](https://github.com/hashicorp/nomad/tree/master/devices/gpu/nvidia).
+
+## Authoring Device Plugins
+
+Authoring a device plugin in Nomad consists of implementing the
+[DevicePlugin][devicePlugin] interface alongside
+a main package to launch the plugin.
+
+The [device plugin skeleton project][skeletonProject] exists to help bootstrap
+the development of new device plugins. It provides most of the boilerplate
+necessary for a device plugin, along with detailed comments.
+
+### Lifecycle and State
+
+A device plugin is long-lived. Nomad will ensure that one instance of the plugin is 
+running. If the plugin crashes or otherwise terminates, Nomad will launch another 
+instance of it. 
+
+However, unlike [task drivers](task-drivers.html), device plugins do not currently
+have an interface for persisting state to the Nomad client. Instead, the device
+plugin API emphasizes fingerprinting devices and reporting their status. After
+helping to provision a task with a scheduled device, they do not have any charter
```

Comment:
I think `charter` here is a bit too complex. Something use something like `access` or `permission`.

Reply:
i was thinking "charter" more in line with "prerogative" or "responsibility". how about the latter?

Reply:
Yes, "responsibility" is good here